RedHatInsights/vulnerability-ui
# VULN-2055 Add tag filter to report config modal

Currently known issue:
- Badge with count of systems with the tag should be right aligned

Jira ticket: https://issues.redhat.com/browse/VULN-2055
Mockup: https://marvelapp.com/prototype/gb738c4/screen/80562239

You can find the ticket by navigating to 'Reports' page, clicking on 'Create report' button in the second card, you can find the tag filter in the middle of the modal.

On insights-qa account there are a lot of tags but not a lot of tags with systems with CVEs, but you can search for tag 'EtzxpiLyrX' and that one has CVEs associated.

Test cases:
- this new filter should not break any other filters, try exporting report with other filters active and check If the CVEs in table match the filter
- you can select tags either from dropdown or modal, check if selecting from either or both of them at the same time works correctly
- try using the search filter, changing pages, changing per page option
- verify whether selected tags are mentioned in the report introduction text (see 4th screenshot)
- try selecting more then 10 tags, only 10 tags and "and {count} more" should be shown in report introduction text
- if global filter has selected tags, inherit the config to the tag filter in modal

![Screenshot from 2022-03-11 15-58-28](https://user-images.githubusercontent.com/8426204/157892262-5988fac4-ed60-42d6-ace1-7c2ecd07589f.png)
![Screenshot from 2022-03-11 15-58-48](https://user-images.githubusercontent.com/8426204/157892267-10da8d06-7863-452f-b7b6-53fbf5808142.png)
![Screenshot from 2022-03-11 15-59-04](https://user-images.githubusercontent.com/8426204/157892271-e152aa01-63e1-4412-ba38-c6f0aa860fc1.png)
![Screenshot from 2022-03-11 16-01-03](https://user-images.githubusercontent.com/8426204/157892557-7090d28a-f645-4a13-8193-8852ef0c2259.png)
